File: python/medium/Longest_Substring_Without_Repeating_Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        answer_list = []
        answer = 0
        for w in s:
            if w not in answer_list:
                answer_list.append(w)
            else:
                if len(answer_list) > answer:
                    answer = len(answer_list)
                answer_list = answer_list[answer_list.index(w)+1:] + [w]
        if len(answer_list) > answer:
                answer = len(answer_list)
        return answer

File: cpp/medium/Longest_Substring_Without_Repeating_Characters.cpp
// #include <string>

// using namespace std;

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if (s.length()==0) return 0;
        int begin = 0;
        int last = 1;
        int answer = 1;
        for (last=1; last < s.length(); ++last){
            for (int i=begin; i < last; ++i){
                if (s[i] == s[last]){
                    begin = i+1;
                    break;
                }
            }
            if (answer < last - begin +1) answer = last - begin +1;
        }
        return answer;
    }
};
